File: 053/cpp/main.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int t = 0;
        int m = nums[0];
        for (auto &num: nums) {
            t += num;
            if (t > 0) {
                m = std::max(m, t);
            } else {
                m = std::max(m, num);
                t = 0;
            }
        }
        return m;
    }
};

File: 053/rust/main.rs
use std::cmp;

impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut t = 0;
        let mut m = nums[0];
        for num in nums {
            t += num;
            if t > 0 {
                m = cmp::max(m, t);
            } else {
                m = cmp::max(m, num);
                t = 0;
            }
        }
        return m;
    }
}
